### PR TITLE
feat: add identifiers to document

### PR DESCRIPTION
**Pull request by baswag**
Depends on https://github.com/koreader/koreader-base/pull/1758

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/11628)

### CODE REVIEW

File: frontend/document/document.lua [hunk 1]
```diff
@@ -209,6 +210,7 @@ function Document:getProps(cached_doc_metadata)
         language     = language,
         keywords     = keywords,
         description  = description,
+        identifiers  = identifiers
```
Comment (Frenzie) [inline frontend/document/document.lua:213]: Sidenote, but we prefer trailing commas (both for less effort when adding something and to at least sometimes have a single-line diff).

File: plugins/coverbrowser.koplugin/bookinfomanager.lua [hunk 2]
```diff
@@ -100,6 +101,7 @@ local BOOKINFO_COLS_SET = {
         "cover_bb_type",
         "cover_bb_stride",
         "cover_bb_data",
+        "identifiers"
```
Comment (hius07) [inline plugins/coverbrowser.koplugin/bookinfomanager.lua:104]: It should be inserted before the "cover" items, to be always accessible in `BookInfoManager:getBookInfo()`.

**Comment by hius07** [comment]
Is it useful to show these id's in the Book information page?
https://github.com/koreader/koreader/blob/c415aea67eee58d846122b07d152e06d25e04121/frontend/apps/filemanager/filemanagerbookinfo.lua#L38-L46

**Comment by poire-z** [comment]
> Is it useful to show these id's in the Book information window?

I hope not :) I think I chose a keyvalueitem per page to get all the bookinfo items on a single page. Adding one item would need me to update it and get other KVPage smaller :)
It's also probably not something one "reads" in there. And copy to be pasted where else?

File: plugins/coverbrowser.koplugin/bookinfomanager.lua [hunk 3]
```diff
@@ -57,6 +57,7 @@ local BOOKINFO_DB_SCHEMA = [[
         language            TEXT,
         keywords            TEXT,
         description         TEXT,
+        identifiers         TEXT,
```
[the remark this answers is not in the document]
Comment (baswag) [reply plugins/coverbrowser.koplugin/bookinfomanager.lua]: Removed it from the cache db

File: plugins/coverbrowser.koplugin/bookinfomanager.lua (hunk 2 above)
Comment (baswag) [reply plugins/coverbrowser.koplugin/bookinfomanager.lua]: Removed from cache db

**Comment by poire-z** [comment]
@hius07 : fine with you this addition to doc_props, and not being used anywhere ? No issue with your various things using/customizing it if we don't (yet) show it or allow customizing it?

@baswag : are you fine with having it only when it's available in the EPUB ? No need to be able to customize/fix it or add one manually from KOReader itself?

I was wondering if there would be something equivalent in FB2 documents, but looking at a few, I see nothing usable (looking for ASCII strings, I expect any such identifiers, equivalent to ISBN for Russian books, to still be digits or ASCII).

There are many possibly valuable (to look at) metadata in both EPUBs and FB2, but they are all (except may be the classic ones we use) inconsistent to be really associated to a keyword. But some cre hooks to extract the metadata XML from an EPUB .opf or an FB2 header could be interesting, just showing some nicely laid out XML. Just some idea for later when/if we ever get bored :)

**Comment by hius07** [comment]
> No issue with your various things using/customizing it if we don't (yet) show it or allow customizing it?

I see no problems. When needed, we iterate over explicitly stated list of properties:
https://github.com/koreader/koreader/blob/6cc970dbb4e181aa2f2e48d7091317cf9973a265/frontend/apps/filemanager/filemanagerbookinfo.lua#L28-L36
But the new property will not be accessible in `ui.doc_props`.

**Pull request merged**